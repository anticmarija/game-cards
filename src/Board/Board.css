.board {
  width: 40%;
  height: 40%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.game-winner {
  font-size: 24px;
  color: yellowgreen;
}

.board-image {
  width: 120px;
  height: auto;
  margin: 0 3px;
  border: 2px solid transparent;
}

.board-winner {
  border: 2px solid red;
  border-radius: 2px;
}

@media only screen and (max-width: 850px) {
  .board-image {
    width: 90px;
  }
}

@media only screen and (max-width: 670px) {
  .board-image {
    width: 70px;
    margin: 0 2px;
  }
}

@media only screen and (max-width: 500px) {
  .board {
    height: 30%;
    flex-wrap: wrap;
  }

  .board-image {
    width: 45px;
    margin: 0 2px;
  }
}
