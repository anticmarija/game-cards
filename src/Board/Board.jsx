import React from 'react';
import { connect } from 'react-redux';

import './Board.css';
import {
  getCardsOnTable,
  getRoundWinnerCard,
} from '../store/reducers/game.reducer';

const Board = ({ cardsOnTable, roundWinnerCard }) => {
  return (
    <div>
      {cardsOnTable.map(card => {
        const winnerStyle = 'board-winner';
        if (roundWinnerCard && roundWinnerCard.user === card.user) {
        }
        return (
          <img
            className={
              roundWinnerCard && roundWinnerCard.user === card.user
                ? winnerStyle
                : ''
            }
            src={card.card.image}
            key={card.card.code}
            alt={card.card.code}
          ></img>
        );
      })}
    </div>
  );
};

export const mapStateToProps = state => ({
  cardsOnTable: getCardsOnTable(state),
  roundWinnerCard: getRoundWinnerCard(state),
});

export default connect(mapStateToProps)(Board);
