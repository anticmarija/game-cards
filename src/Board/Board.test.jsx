import React from 'react';
import { render } from '@testing-library/react';
import { Board, mapStateToProps } from './Board';

describe('Test Board component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Board
        cardsOnTable={[
          { user: 'player1', card: { image: '1', code: '1' } },
          { user: 'player2', card: { image: '2', code: '2' } },
        ]}
        roundWinnerCard={{ user: 'player1', card: { image: '1', code: '1' } }}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
