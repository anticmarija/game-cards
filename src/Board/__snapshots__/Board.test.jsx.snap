// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Board component should render correctly 1`] = `
<div>
  <img
    alt="1"
    class="board-winner"
    src="1"
  />
  <img
    alt="2"
    class=""
    src="2"
  />
</div>
`;
